tiledb/sm/misc/resource_pool.h

```diff
@@ -139,6 +144,79 @@ class ResourcePool {
   friend class ResourceHandle<T, tiledb::sm::ResourcePool>;
 };
 
+template <class T>
+class BlockingResourcePool {
+ public:
+  using resource_handle = ResourceHandle<T, tiledb::sm::BlockingResourcePool>;
+
+  /** Constructor. */
+  BlockingResourcePool(unsigned int n)
+      : resources_(n)
+      , unused_(n)
+      , unused_idx_(n - 1)
+      , num_blocked_threads_(0) {
+    for (unsigned int i = 0; i < n; i++)
+      unused_[i] = i;
+  }
+
+  /** Take a resource from the pool. */
+  resource_handle take() {
+    std::unique_lock<std::mutex> lck(m_);
+    while (resources_exhausted()) {
+      // block until available again
+      num_blocked_threads_++;
+      exhaustion_cv_.wait(lck, [this]() { return !resources_exhausted(); });
```

Comment:
I think the non-predicate form of `wait()` would work here, since it's already being tested as the loop condition.

Reply:
replaced with no-predicate wait